compose/ui/ui-text/src/skikoMain/kotlin/androidx/compose/ui/text/SkiaParagraph.skiko.kt

```diff
@@ -177,29 +178,40 @@ internal class SkiaParagraph(
         return metrics.last()
     }
 
-    override fun getLineHeight(lineIndex: Int) = lineMetrics[lineIndex].height.toFloat()
+    override fun getLineHeight(lineIndex: Int) =
+        if (lineIndex !in lineMetrics.indices) {
```

Comment:
Not sure about this correction here. We can just hide incorrect behavior this way.

Maybe we should move this check to another place? Where the incorrect `lineIndex` is returned?

Reply:
Will it be valid, if we do this check (as `lineCount` returns value always >= 1 now)?
```
if (lineIndex == 0 && lineMetrics.size == 0)
```

Reply:
Yes, I guess it will catch this particular problematic case.

But putting a check for general out of bounds case looks more readable here

Reply:
> But putting a check for general out of bounds case looks more readable here

It is about correctness though. Returning 0 for a wrong index is not correct, we should crash in this scenario, or check calling code. Otherwise we or an user won't notice an error in the other part of code.

Reply:
> if (lineIndex == 0 && lineMetrics.size == 0)

This will work only if there is no case when `lineIndex` can be more than 0.

Also, it seems we have safety checks in the other parts:
<img width="390" alt="image" src="https://user-images.githubusercontent.com/5963351/229594501-494ab15f-7944-4caf-88c6-66be5ce079fb.png">

Not sure, if they were added on purpose.

Reply:
Well, then let's do it like in other places for consistency. Good finding.

Reply:
Ok, let's do this way.

For information:

Skia Paragraph has many issues which we encountered during Compose development. First we were adding hacks in Compose code, and then we created [a fork](https://github.com/JetBrains/skia/commits/skiko), where we should do all fixes now.

In the future, we should probably rewrite Compose implmentation, get rid of all hacks and excessive safety checks.